### ISSUE TITLE
phpcs hang, except in verbose mode!

### ISSUE THREAD

**Issue by Synchro** [squizlabs/PHP_CodeSniffer#2983]
I've got a simple `.phpcs.xml` config file (using phpcs 3.5.5 in PHP 7.4.6):

```xml
<?xml version="1.0"?>
<ruleset xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:noNamespaceSchemaLocation="vendor/squizlabs/php_codesniffer/phpcs.xsd">

  <config name="testVersion" value="7.3-"/>

  <arg name="basepath" value="."/>
  <arg name="cache" value=".phpcs-cache"/>
  <arg name="colors"/>
  <arg name="extensions" value="php"/>
  <arg name="parallel" value="10"/>
  <!-- Show progress -->
  <arg value="p"/>

  <file>src</file>
  <file>tests</file>

  <rule ref="PSR12"/>
  <rule ref="PHPCompatibility"/>
</ruleset>
```

If I run just `phpcs`, it immediately hangs, with no output.
If I run `php -v` it shows this, and then hangs:

```
Registering sniffs in the  standard... DONE (176 sniffs registered)
Creating file list... DONE (1 files in queue)
Loading cache... DONE (0 files in cache)
```
If I run `phpcs -vv`, it produces a load of debug output, but all appears to work and it doesn't hang:

```
[lots of config output]
DONE (5 files in cache)
Changing into directory src
Processing file1.php 
[loaded from cache]... DONE in 0ms (0 errors, 0 warnings)
Processing file2.php 
[loaded from cache]... DONE in 0ms (0 errors, 0 warnings)
Processing file3.php 
[loaded from cache]... DONE in 0ms (0 errors, 0 warnings)
Processing file4.php 
[loaded from cache]... DONE in 0ms (0 errors, 0 warnings)
Processing file5.php 
[loaded from cache]... DONE in 0ms (0 errors, 0 warnings)
Time: 167ms; Memory: 12MB
```

`-vvv` produces even more output but also doesn't hang.

So I'm a bit stuck – the option that shows me debug output also prevents the problem from happening!

**Comment by jrfnl** [comment]
@Synchro This sounds like a duplicate of #2659. which is actually [a PHP issue](https://bugs.php.net/bug.php?id=78717) not a PHPCS issue.

The parallel option is what makes the difference and the problem seems to be related to the `pcntl_waitpid()` function behaviour on MacOS. Turning parallel off should solve it for now.

**Comment by Synchro** [comment]
Thanks for the pointer Juliette. Following those threads led me to find what had gone wrong for me - I recently installed the grpc extension as it is requested by the google API libs, and it appears the bug is in there rather than PHP itself. Sure enough, disabling ext-grpc allows phpcs to work fine in parallel under PHP 7.4.6.

**Comment by jrfnl** [comment]
@Synchro Glad to hear you got it working :+1
